src/redux/auth/index.js

```diff
@@ -0,0 +1,44 @@
+import API from '../api';
+
+export const loadCurrentUser = (state) => state.auth.currentUser;
+
+const LOAD_CURRENT_USER = 'auth/current_user';
+const RESET = 'auth/reset';
+
+export const setCurrentUser = (payload) => ({
+  type: LOAD_CURRENT_USER,
+  payload,
+});
+
+export const login = (payload) => async (dispatch) => {
+  const response = await API.login(payload);
+  dispatch(setCurrentUser(response));
+};
+
+export const logout = () => async (dispatch) => {
+  await API.logout();
+  dispatch(setCurrentUser(null));
+};
+
+export const signup = (payload) => async (dispatch) => {
+  const response = await API.register(payload);
+  dispatch(setCurrentUser(response));
+};
+
+const initialState = {
+  currentUser: null,
+  token: null,
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case LOAD_CURRENT_USER:
+      return action.payload;
```

Comment:
The data coming from the payload is an object with `user`, and `token`. So I would suggest you refactor the return statement to set `currentUser` to `action.payload.user` and `token` to `action.payload`.
```
return {
  currentUser: action.payload.user,
  token: action.payload.token,
};
```